File: algorithm/src/main/java/baekjoon/p01212/Main.java
package baekjoon.p01212;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.next()));
    }

    private static String solve(String input) {
        String[] set = {"000", "001", "010", "011", "100", "101", "110", "111"};
        StringBuilder sb = new StringBuilder();

        for (char ch : input.toCharArray()) {
            sb.append(set[ch - '0']);
        }

        String ret = sb.toString();
        while (ret.charAt(0) == '0' && ret.length() > 1) {
            ret = ret.substring(1);
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p01212/Main.kt
package baekjoon.p01212

fun main(args: Array<String>) {
    println(solve(readLine()!!))
}

private fun solve(input: String): String {
    val set = arrayOf("000", "001", "010", "011", "100", "101", "110", "111")
    val ret = buildString {
        for (ch in input) {
            append(set[ch - '0'])
        }
    }

    val idx = ret.indexOf("1")
    return when (idx) {
        -1 -> "0"
        else -> ret.substring(idx)
    }
}